src/call_context.rs

```diff
@@ -312,4 +336,13 @@ impl<'a> CallContext<'a> {
     pub fn state_mut(&mut self) -> &mut NetworkState {
         &mut self.state
     }
+
+    /// Propagates gas usage to the caller
+    fn gas_merge(&mut self) {
+        let callee_gas_meter = self.gas_meter().clone();
```

Comment:
@ureeves your version does not compile

Reply:
Ah you're right. Immutable borrow while holding a mutable. This could be solved by rearranging things a little. I don't think we should clone the gas meter every time we use `gas_merge`